clientconn_parsed_target_test.go

```diff
@@ -0,0 +1,157 @@
+		{target: "unix-abstract:a-b", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "a-b"}},
+		{target: "unix-abstract:/ a///://::!@#$%^&*()b", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "/ a///://::!@#$%^&*()b"}},
+		{target: "unix-abstract:passthrough:abc", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "passthrough:abc"}},
+		{target: "unix-abstract:unix:///abc", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "unix:///abc"}},
+		{target: "unix-abstract:///a/b/c", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "/a/b/c"}},
+		{target: "unix-abstract:///", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "/"}},
+		{target: "unix-abstract://authority", wantParsed: resolver.Target{Scheme: "unix-abstract", Authority: "", Endpoint: "//authority"}},
+		{target: "passthrough:///unix:///a/b/c", wantParsed: resolver.Target{Scheme: "passthrough", Authority: "", Endpoint: "unix:///a/b/c"}},
+
+		// If we can only parse part of the target.
+		{target: "://", wantParsed: resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "://"}},
+		{target: "unix://domain", wantParsed: resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "unix://domain"}},
+	}
+
+	for _, test := range tests {
+		t.Run(test.target, func(t *testing.T) {
+			cc, err := Dial(test.target, WithInsecure())
+			if err != nil {
+				t.Fatalf("Dial(%q) failed: %v", test.target, err)
+			}
+			defer cc.Close()
+
+			if gotParsed := cc.parsedTarget; gotParsed != test.wantParsed {
+				t.Errorf("cc.parsedTarget = %+v, want %+v", gotParsed, test.wantParsed)
+			}
+		})
+	}
+}
+
+func (s) TestParsedTarget_Failure_WithoutCustomDialer(t *testing.T) {
+	targets := []string{
+		"unix://a/b/c",
+		"unix-abstract://authority/a/b/c",
+	}
+
+	for _, target := range targets {
+		t.Run(target, func(t *testing.T) {
+			if cc, err := Dial(target, WithInsecure()); err == nil {
+				defer cc.Close()
+				t.Fatalf("Dial(%q) succeeded cc.parsedTarget = %+v, expected to fail", target, cc.parsedTarget)
+			}
+		})
+	}
+}
+
+func (s) TestParsedTarget_WithCustomDialer(t *testing.T) {
+	defScheme := resolver.GetDefaultScheme()
+	tests := []struct {
+		target            string
+		wantParsed        resolver.Target
+		wantDialerAddress string
+	}{
+		// unix:[local_path], unix:[/absolute], and unix://[/absolute] have
+		// different behaviors with a custom dialer.
+		{
+			target:            "unix:a/b/c",
+			wantParsed:        resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "unix:a/b/c"},
+			wantDialerAddress: "unix:a/b/c",
+		},
+		{
+			target:            "unix:/a/b/c",
+			wantParsed:        resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "unix:/a/b/c"},
+			wantDialerAddress: "unix:/a/b/c",
+		},
+		{
+			target:            "unix:///a/b/c",
```

Comment:
Should we test a few things that aren't `unix` too?  Maybe `dns:somehostname:port`, e.g.?

Reply:
We don't currently have any special behavior for non-unix schemes when a custom dialer is specified. Are you saying we should add tests for the dns scheme now and make sure things don't break when we make it the default?

Reply:
We probably shouldn't be changing any parsing from what we do today when a custom dialer is involved, not just `unix`.  Maybe we can reason about some changes being okay, but if, e.g., we use the DNS resolver instead of the passthrough resolver, we might break a custom dialer expecting the target name to be provided.

Reply:
Done. Added some tests for dns and "no scheme".